Source/WebGPU/WebGPU/SwapChain.mm

```diff
@@ -28,28 +28,28 @@
 
 #import "APIConversions.h"
 #import "Device.h"
+#import "Surface.h"
 #import "TextureView.h"
 #import <WebGPU/WebGPUExt.h>
 
 namespace WebGPU {
 
-Ref<SwapChain> Device::createSwapChain(WGPUSurface surface, const WGPUSwapChainDescriptor& descriptor)
+Ref<SwapChain> Device::createSwapChain(Surface& surface, const WGPUSwapChainDescriptor& descriptor)
 {
-    return SwapChain::create(WTFMove(surface), descriptor);
+    return surface.createSwapChain(*this, descriptor);
```

Comment:
Having a Surface create a SwapChain seems to invert the ownership relationship. 

Surface -> something that can be rendered to and presented to the screen
SwapChain -> a collection of surfaces which manages presentation to the screen

current implementation of SwapChain has a maximal surface count of 1, that could be lifted if we can resolve memory implications of doing so